pkg/cloudprovider/cloudprovider.go

```diff
@@ -165,12 +168,25 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 	return nc, err
 }
 
-func (c *CloudProvider) waitOnPromise(ctx context.Context, promise *instance.VirtualMachinePromise, nodeClaim *karpv1.NodeClaim) {
+func (c *CloudProvider) waitOnPromise(ctx context.Context, promise *instance.VirtualMachinePromise, nodeClaim *karpv1.NodeClaim, startTime time.Time) {
+	var mainErrForMetrics error
+
 	defer func() {
 		if r := recover(); r != nil {
 			err := fmt.Errorf("%v", r)
 			log.FromContext(ctx).Error(err, "panic during waitOnPromise")
+			mainErrForMetrics = err
 		}
+
+		defer func() {
+			if r := recover(); r != nil {
+				err := fmt.Errorf("%v", r)
+				log.FromContext(ctx).Error(err, "panic during waitOnPromise defer, likely during metrics recording")
+			}
+		}()
+
+		duration := time.Since(startTime)
+		azuremetrics.MethodDurationWithAsync.With(azuremetrics.GetLabelsMapForCloudProviderDurationWithAsync(ctx, "Create", mainErrForMetrics)).Observe(duration.Seconds())
```

Comment:
I think we want these labels to be `snake_case`, so `Create` should be `create`?